waspc/src/Wasp/Psl/Ast/Schema.hs

```diff
@@ -0,0 +1,139 @@
+{-# LANGUAGE DeriveDataTypeable #-}
+
+module Wasp.Psl.Ast.Schema where
+
+import Data.Data (Data)
+
+class IsConfigBlock a where
+  getConfigBlockName :: a -> String
+  getConfigBlockKeyValues :: a -> [ConfigBlockKeyValue]
+
+data Schema = Schema [SchemaElement]
+  deriving (Show, Eq)
+
+data SchemaElement = SchemaModel Model | SchemaEnum PrismaEnum | SchemaDatasource Datasource | SchemaGenerator Generator
+  deriving (Show, Eq)
+
+data Model
+  = Model
+      String
+      -- ^ Name of the model
+      Body
+  deriving (Show, Eq)
+
+newtype Body = Body [Element]
+  deriving (Show, Eq, Data)
+
+data Element = ElementField Field | ElementBlockAttribute Attribute
+  deriving (Show, Eq, Data)
+
+-- TODO: To support attributes before the field,
+--   we could just have `attrsBefore :: [[Attr]]`,
+--   which represents lines, each one with list of attributes.
+data Field = Field
+  { _name :: String,
+    _type :: FieldType,
+    _typeModifiers :: [FieldTypeModifier],
+    _attrs :: [Attribute]
+  }
+  deriving (Show, Eq, Data)
+
+data FieldType
+  = String
+  | Boolean
+  | Int
+  | BigInt
+  | Float
+  | Decimal
+  | DateTime
+  | Json
+  | Bytes
+  | Unsupported String
+  | UserType String
+  deriving (Show, Eq, Data)
+
+data FieldTypeModifier = List | Optional | UnsupportedOptionalList
+  deriving (Show, Eq, Data)
+
+data PrismaEnum
```

Comment:
```
Ambiguous occurrence ‘Enum’
It could refer to
   either ‘Prelude.Enum’,
          imported from ‘Prelude’ at /Users/ilakovac/dev/wasp/waspc/src/Wasp/Psl/Ast/Schema.hs:3:8-26
          (and originally defined in ‘GHC.Enum’)
       or ‘Wasp.Psl.Ast.Schema.Enum’,
          defined at /Users/ilakovac/dev/wasp/waspc/src/Wasp/Psl/Ast/Schema.hs:86:1
```

I didn't want to hide it from Prelude every time we need to use it.

Reply:
Yeah, I would call it Enum, hide import of Prelude.Enum in this specific file, and in other places we will use it qualitifed!

Reply:
Ok you did it I see now! Nice.